### PR TITLE
Support for absolute/relative paths with .. while uploading sources

### PR DESCRIPTION
**Pull request by szymon-kuklewicz**

### CODE REVIEW

File: neptune/projects.py [hunk 1]
```diff
@@ -412,14 +414,25 @@ def create_experiment(self,
             if os.path.isfile(main_file):
                 entrypoint = normalize_file_name(os.path.basename(main_file))
                 upload_source_entries = [
-                    UploadEntry(os.path.abspath(main_file), normalize_file_name(os.path.basename(main_file)))
+                    UploadEntry(abspath(main_file), normalize_file_name(os.path.basename(main_file)))
```
Comment (hubertjaworskineptune) [inline neptune/projects.py]: You imported abspath, but not basename, I think it's confusing and it's better to leave os.path.abspath for reading clarity

File: neptune/projects.py [hunk 2]
```diff
@@ -412,14 +414,25 @@ def create_experiment(self,
...
+                absolute_paths.add(abspath(filepath))
+            try:
+                common_source_root = commonpath(absolute_paths)
+                if os.path.isfile(common_source_root):
+                    common_source_root = os.path.dirname(common_source_root)
+                for absolute_path in absolute_paths:
+                    upload_source_entries.append(UploadEntry(absolute_path, normalize_file_name(os.path.relpath(absolute_path, common_source_root))))
+            except ValueError:
+                for absolute_path in absolute_paths:
+                    upload_source_entries.append(UploadEntry(absolute_path, normalize_file_name(absolute_path)))
```
Comment (hubertjaworskineptune) [inline neptune/projects.py]: 1. When error occurs on last file you re-upload them all
2.Where exactly does the ValueError come from?

File: neptune/internal/storage/storage_utils.py [hunk 3]
```diff
@@ -134,7 +134,7 @@ def split_upload_files(upload_entries, max_package_size=1 * 1024 * 1024, max_fil
 
 
 def normalize_file_name(name):
-    return name.replace(os.sep, '/')
+    return name.replace(sep, '/')
```
Comment (hubertjaworskineptune) [inline neptune/internal/storage/storage_utils.py]: I think this code is confusing long-term - sep looks like a normal variable and I need to look up imports to understand what it is

File: neptune/projects.py (hunk 2 above)
Comment (szymon-kuklewicz) [reply neptune/projects.py]: This only collects UploadEntry in the memory, does not send files anywhere. Actual sending is done later, out of scope of this change.

**Pull request merged**